SasquatchMac/SasquatchMac/ViewControllers/TransmissionViewController.swift

```diff
@@ -0,0 +1,424 @@
+          cell.subviews[1].isHidden = true
+          return cell
+        case kAppLocaleRow:
+          let key: NSTextField = cell.subviews[0] as! NSTextField
+          key.stringValue = property.key
+          let value: NSTextField = cell.subviews[2] as! NSTextField
+          value.stringValue = property.value
+          cell.subviews[1].isHidden = true
+          return cell
+        case kUserIdRow:
+          let key: NSTextField = cell.subviews[0] as! NSTextField
+          key.stringValue = property.key
+          let value: NSTextField = cell.subviews[2] as! NSTextField
+          value.stringValue = property.value
+          cell.subviews[1].isHidden = true
+          return cell
+        default:
+           return nil
+        }
+      }
+      return nil
+    }
+
+    //Transmission target section
+    if let cell = tableView.make(withIdentifier: "target", owner: nil) as? NSTableCellView {
+      let section = transmissionTargetSections![tableView.tag]
+      switch (row) {
+      case kEnabledCellRowIndex:
+        let key: NSTextField = cell.subviews[0] as! NSTextField
+        key.stringValue = "Set Enabled"
+        let value: NSButton = cell.subviews[1] as! NSButton
+        value.state = section.isTransmissionTargetEnabled().hashValue
+        value.isEnabled = tableView.tag != Section.Default.rawValue
+        value.target = self
+        value.action = #selector(targetEnabledSwitchValueChanged)
+        cell.subviews[2].isHidden = true
+        return cell     
+      case kAnalyticsCellRowIndex:
+        let key: NSTextField = cell.subviews[0] as! NSTextField
+        key.stringValue = "Analytics Events"
+        let value: NSButton = cell.subviews[1] as! NSButton
+        value.state = section.shouldSendAnalytics().hashValue
+        value.target = self
+        value.action = #selector(targetShouldSendAnalyticsSwitchValueChanged)
+        cell.subviews[2].isHidden = true
+        return cell
+      case kTokenCellRowIndex:
+        let key: NSTextField = cell.subviews[0] as! NSTextField
+        key.stringValue = "Token"
+        let value: NSTextField = cell.subviews[2] as! NSTextField
+        value.stringValue = section.token!
+        cell.subviews[1].isHidden = true
+        return cell
+      case kPauseCellRowIndex:
+        cell.subviews[0].isHidden = true
+        cell.subviews[1].isHidden = true
+        cell.subviews[2].isHidden = true
+        let pause: NSButton = cell.subviews[3] as! NSButton
+        pause.isHidden = false
+        pause.action = #selector(TransmissionViewController.pause)
+        let resume: NSButton = cell.subviews[4] as! NSButton
+        resume.isHidden = false
+        resume.action = #selector(TransmissionViewController.resume)
+        return cell
+      default:
+        return nil
+      }
+    }
+    return nil
+  }
+
+  private func getCellSection(forView view: NSView) -> Int {
+    guard let tableView = view.superview!.superview?.superview as? NSTableView else {
```

Comment:
using `view.superview.superview` etc. looks like it would break if the layout ever changes... I feel like there's a better way to do this but I'm not sure

Reply:
I refer to the iOS Sasquatch app code to write this. And I don't have a better way to replace it currently.